sci_analysis/test/test_graph_groupscatter.py

```diff
@@ -372,6 +372,48 @@ def test_27_scatter_two_groups_negative_corr(self):
         self.assertTrue(GraphGroupScatter(input_array['a'], input_array['b'], groups=input_array['c'],
                                           save_to='{}test_group_scatter_27'.format(self.save_path)))
 
+    def test_28_scatter_two_groups_labels(self):
+        np.random.seed(987654321)
+        input_1_x = st.norm.rvs(size=100)
+        input_1_y = [x + st.norm.rvs(0, 0.5, size=1)[0] for x in input_1_x]
+        input_2_x = st.norm.rvs(size=100)
+        input_2_y = [(x / 2) + st.norm.rvs(0, 0.2, size=1)[0] for x in input_2_x]
+        input_labels_array = np.random.choice(list('ABCDE'),  size=(200))
+        grp = [1] * 100 + [2] * 100
+        cs_x = np.concatenate((input_1_x, input_2_x))
+        cs_y = np.concatenate((input_1_y, input_2_y))
+        input_array = pd.DataFrame({'a': cs_x, 'b': cs_y, 'c': grp})
+        self.assertTrue(GraphGroupScatter(input_array['a'], input_array['b'], groups=input_array['c'], labels=input_labels_array,
+                                          save_to='{}test_group_scatter_28'.format(self.save_path)))
+
+    def test_29_scatter_two_groups_labels_missing_data(self):
+        np.random.seed(987654321)
+        input_1_x = st.norm.rvs(size=100)
+        input_1_y = [x + st.norm.rvs(0, 0.5, size=1)[0] for x in input_1_x]
+        input_2_x = st.norm.rvs(size=100)
+        input_2_y = [(x / 2) + st.norm.rvs(0, 0.2, size=1)[0] for x in input_2_x]
+        input_labels_array = np.random.choice(list('ABCDE'),  size=(220))
+        grp = [1] * 110 + [2] * 110
+        cs_x = np.concatenate((input_1_x, input_2_x))
+        cs_y = np.concatenate((input_1_y, input_2_y))
+        indicies_x = list(np.random.randint(0, 199, 20))
+        indicies_y = list(np.random.randint(0, 199, 20))
+        for i in indicies_x:
+            cs_x = np.insert(cs_x, i, np.nan, axis=0)
+        for i in indicies_y:
+            cs_y = np.insert(cs_y, i, np.nan, axis=0)
+        input_array = pd.DataFrame({'a': cs_x, 'b': cs_y, 'c': grp})
+        self.assertTrue(GraphGroupScatter(input_array['a'], input_array['b'], groups=input_array['c'], labels=input_labels_array,
+                                          save_to='{}test_group_scatter_29'.format(self.save_path)))
+
+
+    def test_30_groupscatter_dataframe(self):
+        """tests graphscater with dataframe input"""
+        np.random.seed(987654321)
+        df = pd.DataFrame(np.random.randn(100, 2), columns=list('xy'))
+        df['labels'] = np.random.choice(list('ABCDE'), len(df)).tolist()
+        df['groups'] = np.random.choice(list('XYZ'), len(df)).tolist()
+        self.assertTrue(GraphGroupScatter(df['x'], df['y'], groups=df['groups'], labels= df['labels'], save_to='{}test_scatter_135'.format(self.save_path)))
```

Comment:
The filename in save_to doesn't match the naming convention in the other tests (test_group_scatter_30).